app/src/main/java/com/example/adhirajdatar/mdbsocials/UserFeedActivity.java

```diff
@@ -0,0 +1,109 @@
+package com.example.adhirajdatar.mdbsocials;
+
+import android.content.Intent;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class UserFeedActivity extends AppCompatActivity {
+
+    private FirebaseAuth mAuth;
+    UserFeedAdapter adapter;
+    RecyclerView eventsListView;
+    DatabaseReference mDatabase;
+
+    ArrayList<Social> socials = new ArrayList<Social>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_user_feed);
+
+        mAuth = FirebaseAuth.getInstance();
+        mDatabase = FirebaseDatabase.getInstance().getReference();
+
+        eventsListView = (RecyclerView) findViewById(R.id.eventsFeedRecyclerView);
+        eventsListView.setLayoutManager(new LinearLayoutManager(this));
+        adapter = new UserFeedAdapter(getApplicationContext(), socials);
+        eventsListView.setAdapter(adapter);
+
+        ((FloatingActionButton)findViewById(R.id.addEventButton)).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(getApplicationContext(),AddSocial.class);
+                intent.putExtra("amtSocials", socials.size());
+                startActivity(intent);
+            }
+        });
+
+        mDatabase.child("Socials").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                socials = new ArrayList<Social>();

```

Comment:
it's better practice to define this outside, and then call .clear and notify data set changed. This way, you won't have to worry about calling methods within the adapter